skore/src/skore/cross_validate.py

```diff
@@ -0,0 +1,235 @@
+        A list of scorers in `cv_results` which should be removed.
+
+    Returns
+    -------
+    dict
+        A new cv_results dict, with the specified scorers information removed.
+    """
+    # Takes care both of train and test scores
+    return {
+        k: v
+        for k, v in cv_results.items()
+        if not any(added_scorer in k for added_scorer in added_scorers)
+    }
+
+
+def cross_validate(
+    *args, project: Optional[Project] = None, **kwargs
+) -> CrossValidationItem:
+    """Evaluate estimator by cross-validation and output UI-friendly object.
+
+    This function wraps scikit-learn's [cross_validate](https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.cross_validate.html#sklearn.model_selection.cross_validate)
+    function, to provide more context
+    and facilitate the analysis.
+    As such, the arguments are the same as scikit-learn's cross_validate function.
+
+    Parameters
+    ----------
+    The same parameters as scikit-learn's cross_validate function, except for
+
+    project : Project or None
+        A project to save cross-validation data into. If None, no save is performed.
+
+    Returns
+    -------
+    cv_results : dict
+        A dict of the form returned by scikit-learn's cross_validate function.
+
+    Examples
+    --------
+    >>> def prepare_cv():
+    ...     from sklearn import datasets, linear_model
+    ...     diabetes = datasets.load_diabetes()
+    ...     X = diabetes.data[:150]
+    ...     y = diabetes.target[:150]
+    ...     lasso = linear_model.Lasso()
+    ...     return lasso, X, y
+
+    >>> project = skore.load("project.skore")  # doctest: +SKIP
+    >>> lasso, X, y = prepare_cv()  # doctest: +SKIP
+    >>> cross_validate(lasso, X, y, cv=3, project=project)  # doctest: +SKIP
+    alt.Chart(...)
+    {'fit_time': array(...), 'score_time': array(...), 'test_score': array(...)}
+    """
+    import sklearn.model_selection
+
+    # Recover specific arguments
+    estimator = args[0] if len(args) >= 1 else kwargs.get("estimator")
+    X = args[1] if len(args) >= 2 else kwargs.get("X")
+    y = args[2] if len(args) == 3 else kwargs.get("y")
+
+    try:
+        scorers = kwargs.pop("scoring")
+    except KeyError:
+        scorers = None
+
+    # Extend scorers with other relevant scorers
+    new_scorers, added_scorers = _add_scorers(estimator, y, scorers)
+
+    cv_results = sklearn.model_selection.cross_validate(
+        *args, **kwargs, scoring=new_scorers
+    )
+    # Add explicit metric to result (rather than just "test_score")
+    if isinstance(scorers, str):
+        if kwargs.get("return_train_score") is not None:
+            cv_results[f"train_{scorers}"] = cv_results["train_score"]
+        cv_results[f"test_{scorers}"] = cv_results["test_score"]
+
+    cross_validation_item = CrossValidationItem.factory(cv_results, estimator, X, y)
+
+    if project is not None:
+        project.put_item("cross_validation", cross_validation_item)
```

Comment:
Question: how will someone compare two different cross validation evaluations?

Is it planned to offer a way to name or prefix the generated item keys?

cc @MarieS-WiMLDS

Reply:
it's the second part of the feature to be able to compare yes indeed. At first, we will ask the user to say if it's comparable, otherwise we will consider that they are always comparable inside the same project. Each run will be named run_{n}, and will have a "info" chart that can be displayed on hoover, containing: the estimator name, number of cv, id of the data used (to be defined yet, displaying an hash is not very user-friendly).